torch/nn/_functions/thnn/batchnorm_double_backwards.py

```diff
@@ -0,0 +1,97 @@
+from torch.autograd.variable import Variable
+from functools import reduce
+from operator import mul
+
+
+def sum_exclude_dim1(to_sum, keepdim=True):
+    to_sum = to_sum.sum(dim=0, keepdim=True)
+    dim = 2
+    for dim in range(2, to_sum.dim()):
+        to_sum = to_sum.sum(dim=dim, keepdim=True)
+    return to_sum
+
+
+# because gamma/ggG/ggB are 1-dimensional and represent dim==1, we can't
+# do a straight expansion because it won't follow the broadcasting rules.
+def expand_as_dim1(src, target):
+    src_expanded = src
+    while len(src_expanded.size()) < len(target.size()) - 1:
+        src_expanded = src_expanded.unsqueeze(1)
+    return src_expanded.expand_as(target)
+
+
+def batchnorm_double_backwards_fn(input, gamma, ggI, ggG, ggB, gO, eps):
+    affine = gamma is not None
+    if affine:
+        gamma_expanded = expand_as_dim1(gamma, input)
+
+        if ggG is not None:
+            ggG_expanded = expand_as_dim1(ggG, input)
+
+        if ggB is not None:
+            ggB_expanded = expand_as_dim1(ggB, input)
+    else:
+        gamma_expanded = 1
+
+    # define some terms we will reuse
+    M = reduce(mul, input.size()[0:1] + input.size()[2:])
+    mu = sum_exclude_dim1(input).div_(M)
+    input_sub_mu = input - mu
```

Comment:
It could, but from my reading of the API it would require two steps: i.e. mu.sub_(input).mul_(-1).